internal/cli/internal/command/command.go

```diff
@@ -408,21 +408,15 @@ func RequireAppName(ctx context.Context) (context.Context, error) {
 // LoadAppNameIfPresent is a Preparer which adds app name if the user has used --app or there appConfig
 // but unlike RequireAppName it does not error if the user has not specified an app name.
 func LoadAppNameIfPresent(ctx context.Context) (context.Context, error) {
-	ctx, err := LoadAppConfigIfPresent(ctx)
-	if err != nil {
-		return nil, err
-	}
+	ctx, err := RequireAppName(ctx)
 
-	name := flag.GetApp(ctx)
-	if name == "" {
-		// if there's no flag present, first consult with the environment
-		if name = env.First("FLY_APP"); name == "" {
-			// and then with the config file (if any)
-			if cfg := app.ConfigFromContext(ctx); cfg != nil {
-				name = cfg.AppName
-			}
+	if err != nil {
+		if errors.Is(err, errRequireAppName) {
+			return ctx, nil
```

Comment:
if you get an error you should set the name to an empty string. here you're returning a context which doesn't carry a name so callers of `app.NameFromContext` will crash.

Reply:
You are right I should have tested that path